framework/base/Module.php

```diff
@@ -649,6 +648,23 @@ public function createControllerByID($id)
         return null;
     }
 
+    /**
+     * @param $className
+     * @param $prefix
+     * @return bool
+     */
+    private function incorrectClassNameOrPrefix($className, $prefix)
+    {
+        if (!preg_match('%^[a-z][a-z0-9\\-_]*$%', $className)) {
```

Comment:
ClassName can be in format CamelCase?

Reply:
No, it is part of route and Yii2 splits CamelCase to dashed names.